airbyte-integrations/connectors/source-gitlab/source_gitlab/source.py

```diff
@@ -0,0 +1,122 @@
+
+        auth = TokenAuthenticator(token=config["private_token"])
+        auth_params = dict(authenticator=auth, api_url=config["api_url"])
+        groups = Groups(group_ids=gids, **auth_params)
+        if gids:
+            projects = Projects(project_ids=pids, parent_stream=groups, **auth_params)
+        else:
+            projects = Projects(project_ids=pids, **auth_params)
+
+        return groups, projects
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        try:
+            groups, projects = self._generate_main_streams(config)
+            for stream in projects.stream_slices(sync_mode=SyncMode.full_refresh):
+                list(projects.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream))
+            return True, None
+        except Exception as error:
+            return False, f"Unable to connect to Gitlab API with the provided credentials - {repr(error)}"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = TokenAuthenticator(token=config["private_token"])
+        auth_params = dict(authenticator=auth, api_url=config["api_url"])
+
+        groups, projects = self._generate_main_streams(config)
+        pipelines = Pipelines(parent_stream=projects, start_date=config["start_date"], **auth_params)
+        merge_requests = MergeRequests(parent_stream=projects, start_date=config["start_date"], **auth_params)
+
+        streams = [
+            groups,
+            projects,
+            Branches(parent_stream=projects, repository_part=True, **auth_params),
+            Commits(parent_stream=projects, repository_part=True, start_date=config["start_date"], **auth_params),
+            Issues(parent_stream=projects, start_date=config["start_date"], **auth_params),
+            Jobs(parent_stream=pipelines, **auth_params),
+            Milestones(parent_stream=projects, parent_similar=True, **auth_params),
+            Milestones(parent_stream=groups, parent_similar=True, **auth_params),
+            Members(parent_stream=projects, parent_similar=True, **auth_params),
+            Members(parent_stream=groups, parent_similar=True, **auth_params),
+            Labels(parent_stream=projects, parent_similar=True, **auth_params),
+            Labels(parent_stream=groups, parent_similar=True, **auth_params),
+            merge_requests,
+            Releases(parent_stream=projects, **auth_params),
+            Tags(parent_stream=projects, repository_part=True, **auth_params),
+            pipelines,
+            Users(parent_stream=projects, **auth_params),
+        ]
+
+        # Enabling additional streams according to the setting
+        if config["ultimate_license"]:
+            epics = Epics(parent_stream=groups, **auth_params)
+            streams += [epics, EpicIssues(parent_stream=epics, **auth_params)]
+
+        if config["fetch_merge_request_commits"]:
```

Comment:
why do we need this config option? use can always uncheck the stream in the UI

Reply:
It remained a vestige of Singer, I myself do not understand why I left it, if we allow the user to choose streams.
Fixed